docs/upgrading.md

```diff
@@ -1,66 +1,104 @@
+There are some further requirements for this, see the [howto](#enabling-leader-election) section below for more.
 
-There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
+### Enabling TLS for gRPC
+
+There is a Job in the `ds-hostnet-split` directory that will use the new `contour certgen` command
+to generate a CA and then sign Contour and Envoy keypairs, which can also then be saved directly
+to Kubernetes as Secrets, ready to be mounted into your Contour and Envoy Deployments and Daemonsets.
 
 Again, check the examples for Contour and Envoy in `ds-hostnet-split` for the exact commands.
 
-The steps you will need to follow for this to work are detailed more in (grpc-tls-howto.md)[./grpc-tls-howto.md]:
+If you wish to do this manually, you can use the steps detailed in (grpc-tls-howto.md)[./grpc-tls-howto.md].
+
+In short, you will need to:
 1. Generate a CA keypair
-1. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
-1. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
-1. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+2. Use it to generate and sign two more keypairs, one for Contour and one for Envoy.
+3. Get the CA certificate (only) and the Contour and Envoy keypairs into Kubernetes Secrets inside the `heptio-contour` (or whatever) namespace.
+5. Mount the CA cert and the keypair into the Contour and Envoy pods, and supply the correct parameters to `contour serve` (for Contour) and `contour bootstrap` (for Envoy)
+
+As mentioned in (grpc-tls-howto.md)[./grpc-tls-howto.md], the `contour certgen` command can
+perform some or all of the changes for you.
 
-If you are interested in the exact details of how the certificates need to be generated, see `docs/grpc-tls-howto.md`, which explains the `openssl` commands required to perform the same steps as `contour certgen` can do for you.
+### Enabling Leader Election
 
+To enable leader election, the following must be true for you:
+- You are running in a split Contour and Envoy setup. That is, there are separate Contour
+and Envoy pod(s). Note that leader election will still work if there is only one Contour pod.
+- The Contour must have a TCP readiness probe configured to check port 8001 (the gRPC port),
+as non-leaders will not serve gRPC, and Envoys may not be properly configured if they attempt
+to connect to a non-leader Contour.
+- The update strategy for the Contour deployment must be changed to `Recreate` instead of
```

Comment:
Same as above, its a change to be made to enable leader election, not a check to decide if you should use leader election.